core/unix/module.c

```diff
@@ -72,6 +72,8 @@ os_module_area_init(module_area_t *ma, app_pc base, size_t view_size, bool at_ma
     char *soname = NULL;
     ASSERT(module_is_header(base, view_size));
 
+    ma->os_data.source_file_name = filepath;
```

Comment:
> This is not safe to store just a pointer permanently, since filepath's lifetime is not guaranteed beyond this function. If the idea is just to use this during this function, it should be cleared afterward (with a comment in the struct): but IMHO it's cleaner to not have such a field and instead pass extra parameters.

This was my first thought in the initial change but `module_walk_program_headers()` is called from 10 places and that's quite an intrusive change. Also, it already has 9 parameters!

> However, we already have the path in module_area_t.full_path stored down below. I guess these os_module_info routines aren't passed the module_area_t is the problem? One idea is to store a backpointer in os_module_data_t to module_area_t here, and move the full_path assignment up here.

I prefer this approach but the backpointer to `module_area_t` in `os_module_data_t` caused recursive #include issues in the build.

So, the change I've made is to avoid using `filepath` and set `source_file_name` to `module_area_t`'s `full_path`. I've also changed the name of `source_file_name` to `source_file_path`.

Reply:
OK.